README.md

```diff
@@ -177,12 +177,20 @@ The client specified MUST implement functions called `getObject` and `putObject`
 
 Sets the `Cache-Control` header on the uploaded files.
 
+This can also be a function. The function will be called for each file, with the
```

Comment:
@martndemus Instead of duplicating, how would you feel about this:

```js
const S3_DEFAULTS = {
  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
  bucket: process.env.S3_BUCKET_NAME,
  region: process.env.S3_BUCKET_REGION
};

const ENV = {
  'assets-s3': Object.assign({}, S3_DEFAULTS, {
    fileIgnorePattern: 'sw.js',
  }),

  'sw-s3': Object.assign({}, S3_DEFAULTS, {
    filePattern: 'sw.js'
    cacheControl: 'no-cache'
  })
};
```

One thing here is that this plugin does not currently implement `fileIgnorePattern` but totally could, just as [ember-cli-deploy/ember-cli-deploy-manifest](https://github.com/ember-cli-deploy/ember-cli-deploy-manifest) does